addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2523,6 +2523,15 @@ const SnippetOptionWidget = Widget.extend({
                         values = values.join(' ').replace(color, '').trim();
                         value = `${color} ${values}${inset ? ' inset' : ''}`;
                     }
+                    if (params.cssProperty === 'border-width' && value.endsWith('px')) {
+                        if (this.$target[0].style[params.cssProperty]) {
+                            // If defined: trust style attribute.
+                            value = this.$target[0].style[params.cssProperty];
+                        } else {
+                            // Rounding value up avoids zoom-in issues.
+                            value = `${Math.ceil(parseFloat(value))}px`;
+                        }
+                    }
```

Comment:
Code is ok for me and it works :+1: 
I don't know why we don't rely on the HTML from the beginning. I also like another solution which consists in taking into account the zoom of the screen with this code.
`Math.round(window.devicePixelRatio * 100)`
But I don't have any argument to say that it would be better than the solution you propose

Reply:
I had a look at `devicePixelRatio` and I could not use it: it combines both the browser zoom and the OS zoom (on some platforms at least).
Then when I saw the mapping between set values and the ones obtained with `getComputedStyle`, I could not find a relationship between the two number that would involve `devicePixelRatio`.

Reply:
> I had a look at devicePixelRatio and I could not use it: it combines both the browser zoom and the OS zoom (on some platforms at least).

Note that on my laptop, I have a system setting to enlarge everything, indeed devicePixelRatio detect it and it causes the issue (even without chrome zoom).

> Then when I saw the mapping between set values and the ones obtained with getComputedStyle, I could not find a relationship between the two number that would involve devicePixelRatio.

Ok, so I suppose your solution is the good one :+1:

Reply:
> - Set the border of a column to 10 px
>
> => When you press enter or leave the input the value change to 9.X

I am not really sure I would consider that to be a bug TBH: that's kinda the base of the website builder. The user edits value but the padel displayed values are the one actually used by the browser. I agree it is a bit strange in this specific case but is it worth a specific fix?

In any case the solution seems dubious to me:
- Is this really a "bug" confined to border widths? (not all pixel values?)
- Relying on the style attribute may be ok (although I would prefer if we do not have to do that) but it is inexact. The point of using the computed style... is to have the right one. Checking only the inline style is only 100% sure if it is set as "important" :grimacing: If needed, maybe it is worth comparing the inline style with the computed style?
- If we do not want to display sub-pixel values, I would rather do that only and explicitly, no? (and in that case why not `round` instead of `ceil`?)

Reply:
> Is this really a bug confined to border-widths ?

As far as I can tell, yes.

Running the following code (e.g. on odoo.com - but I get the same result on other pages):
```js
find = () => {
    const properties = new Set();
    for (const el of document.body.querySelectorAll("*")) {
        style = getComputedStyle(el);
        for (const property of style) {
            const oldValue = style[property];
            el.setAttribute("style", `${property}: 17px; !important`);
            const value = style[property];
            el.setAttribute("style", "");
            if (value !== oldValue && value.endsWith('px') && value !== '17px') {
                properties.add(property);
            }
        }
    }
    return properties;
}
// Switch to 100%
found100 = find();
// Switch to 125% in Chrome or 120% in Firefox
found125 = find();
[...found125].filter(value => !found100.has(value));
```
I get the following output:
```js
[
    // Chrome only:
    "tab-size",
    "-webkit-border-horizontal-spacing",
    "-webkit-border-vertical-spacing",
    // Chrome and Firefox:
    "border-block-start-width",
    "border-top-width",
    "border-block-end-width",
    "border-bottom-width",
    "border-inline-end-width",
    "border-inline-start-width",
    "border-left-width",
    "border-right-width"
]
```

Reply:
> If we do not want to display sub-pixel values, I would rather do that only and explicitly, no?

:+1: I did it because the rounding approach does not work for zoom-out... but I do not really think that editing the web page while zoomed out makes any sense.

> (and in that case why not `round` instead of `ceil`?)

Because it is the one that returns the wanted value based on the observed values (see the table in the commit message)
E.g.:
```
Value  Chrome 125%  Firefox 120%
  7px        6.4px     6.66667px
  9px        8.8px     8.33333px
```
`round(6.4)` would give 6, `round(8.33333)` would give 8.

Reply:
@qsm-odoo Adapted code to not rely on style attribute and updated commit message accordingly.
Re-tested on runbot with Chrome & Firefox, zoomed in and non-zoomed.